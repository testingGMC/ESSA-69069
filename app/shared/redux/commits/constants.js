/*
 *
 * Commits constants
 *
 */

export const FETCH_COMMITS_REQUEST = 'app/Commits/FETCH_COMMITS_REQUEST'
export const FETCH_COMMITS_SUCCESS = 'app/Commits/FETCH_COMMITS_SUCCESS'
export const FETCH_COMMITS_FAILURE = 'app/Commits/FETCH_COMMITS_FAILURE'

export const FETCH_LATEST_COMMITS_REQUEST = 'app/Commits/FETCH_LATEST_COMMITS_REQUEST'
export const FETCH_LATEST_COMMITS_SUCCESS = 'app/Commits/FETCH_LATEST_COMMITS_SUCCESS'
export const FETCH_LATEST_COMMITS_FAILURE = 'app/Commits/FETCH_LATEST_COMMITS_FAILURE'

export const REVERT_COMMIT_REQUEST = 'app/Commits/REVERT_COMMIT_REQUEST'
export const REVERT_COMMIT_SUCCESS = 'app/Commits/REVERT_COMMIT_SUCCESS'
export const REVERT_COMMIT_FAILURE = 'app/Commits/REVERT_COMMIT_FAILURE'