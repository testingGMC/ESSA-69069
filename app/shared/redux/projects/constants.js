/*
 *
 * Projects constants
 *
 */

export const FETCH_PROJECTS_REQUEST = 'app/Projects/FETCH_PROJECTS_REQUEST'
export const FETCH_PROJECTS_SUCCESS = 'app/Projects/FETCH_PROJECTS_SUCCESS'
export const FETCH_PROJECTS_FAILURE = 'app/Projects/FETCH_PROJECTS_FAILURE'

export const ADD_PROJECT_REQUEST = 'app/Projects/ADD_PROJECT_REQUEST'
export const ADD_PROJECT_SUCCESS = 'app/Projects/ADD_PROJECT_SUCCESS'
export const ADD_PROJECT_FAILURE = 'app/Projects/ADD_PROJECT_FAILURE'

export const DELETE_PROJECT_REQUEST = 'app/Projects/DELETE_PROJECT_REQUEST'
export const DELETE_PROJECT_SUCCESS = 'app/Projects/DELETE_PROJECT_SUCCESS'
export const DELETE_PROJECT_FAILURE = 'app/Projects/DELETE_PROJECT_FAILURE'

export const UPDATE_PROJECT_REQUEST = 'app/Projects/UPDATE_PROJECT_REQUEST'
export const UPDATE_PROJECT_SUCCESS = 'app/Projects/UPDATE_PROJECT_SUCCESS'
export const UPDATE_PROJECT_FAILURE = 'app/Projects/UPDATE_PROJECT_FAILURE'

export const SET_IS_UPDATE_MODAL_VISIBLE_REQUEST = 'app/Projects/SET_IS_UPDATE_MODAL_VISIBLE_REQUEST'
export const SET_IS_UPDATE_MODAL_VISIBLE_SUCCESS = 'app/Projects/SET_IS_UPDATE_MODAL_VISIBLE_SUCCESS'
export const SET_IS_UPDATE_MODAL_VISIBLE_FAILURE = 'app/Projects/SET_IS_UPDATE_MODAL_VISIBLE_FAILURE'


export const SET_IS_VISIBLE_REQUEST = 'app/Projects/SET_IS_VISIBLE_REQUEST'
export const SET_IS_VISIBLE_SUCCESS = 'app/Projects/SET_IS_VISIBLE_SUCCESS'
export const SET_IS_VISIBLE_FAILURE = 'app/Projects/SET_IS_VISIBLE_FAILURE'
