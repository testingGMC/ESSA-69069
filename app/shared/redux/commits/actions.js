/*
 *
 * Commits actions
 *
 */
import * as CONSTANTS from './constants'

export function fetchCommits() {
  return {
    type: CONSTANTS.FETCH_COMMITS_REQUEST,
  }
}

export function fetchLatestCommits() {
  return {
    type: CONSTANTS.FETCH_LATEST_COMMITS_REQUEST,
  }
}

export function revertToACommit(id, message, description) {
  return {
    type: CONSTANTS.REVERT_COMMIT_REQUEST,
    id,
    message,
    description,
  }
}
