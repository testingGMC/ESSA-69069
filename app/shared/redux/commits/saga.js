import { put, takeEvery, all, call } from 'redux-saga/effects'
import * as api from 'shared/services/version-control.service'
import { push } from 'connected-react-router'
import * as CONSTANTS from './constants'

export function* fetchCommits() {
  try {
    const data = yield call(api.getCommits)

    yield put({
      type: CONSTANTS.FETCH_COMMITS_SUCCESS,
      commits: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_COMMITS_FAILURE, e })
  }
}
export function* fetchLatestCommits() {
  try {
    const data = yield call(api.getLatestCommits)
    yield put({
      type: CONSTANTS.FETCH_LATEST_COMMITS_SUCCESS,
      commits: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_LATEST_COMMITS_FAILURE, e })
  }
}
export function* revertToACommit(action) {
  try {
    const data = yield call(
      api.revertToACommit,
      action.id,
      action.message,
      action.description,
    )
    yield put({
      type: CONSTANTS.REVERT_COMMIT_SUCCESS,
      commit: data.model,
    })
    yield put(push('/'))
  } catch (e) {
    yield put({ type: CONSTANTS.REVERT_COMMIT_FAILURE, e })
  }
}

export function* fetchCommitsWatcher() {
  yield takeEvery(CONSTANTS.FETCH_COMMITS_REQUEST, fetchCommits)
}

export function* fetchLatestCommitsWatcher() {
  yield takeEvery(CONSTANTS.FETCH_LATEST_COMMITS_REQUEST, fetchLatestCommits)
}

export function* revertCommitsWatcher() {
  yield takeEvery(CONSTANTS.REVERT_COMMIT_REQUEST, revertToACommit)
}

export default function* commitsSaga() {
  yield all([
    fetchCommitsWatcher(),
    fetchLatestCommitsWatcher(),
    revertCommitsWatcher(),
  ])
}
