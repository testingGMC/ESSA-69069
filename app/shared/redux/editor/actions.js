/*
 *
 * Editor actions
 *
 */
import * as CONSTANTS from './constants'

export function clearState() {
  return {
    type: CONSTANTS.CLEAR_STATE_SUCCESS,
  }
}

export function fetchSkill(skillId) {
  return {
    type: CONSTANTS.FETCH_SKILL_REQUEST,
    skillId,
  }
}
export function updateSkill(skillId, skill) {
  return {
    type: CONSTANTS.UPDATE_SKILL_REQUEST,
    skillId,
    skill,
  }
}

export function addLearningStyle(learningStyle) {
  return {
    type: CONSTANTS.ADD_LEARNING_STYLE_REQUEST,
    learningStyle,
  }
}

export function deleteLearningStyle(learningStyleId) {
  return {
    type: CONSTANTS.DELETE_LEARNING_STYLE_REQUEST,
    learningStyleId,
  }
}

export function addPage(page, learningStyleId) {
  return {
    type: CONSTANTS.ADD_PAGE_REQUEST,
    page,
    learningStyleId,
  }
}

export function updatePage(pageId, page, learningStyleId) {
  return {
    type: CONSTANTS.UPDATE_PAGE_REQUEST,
    pageId,
    page,
    learningStyleId,
  }
}

export function deletePage(pageId, learningStyleId) {
  return {
    type: CONSTANTS.DELETE_PAGE_REQUEST,
    pageId,
    learningStyleId,
  }
}

export function addAssessment(assessment) {
  return {
    type: CONSTANTS.ADD_ASSESSMENT_REQUEST,
    assessment,
  }
}

export function deleteAssessment(assessmentId) {
  return {
    type: CONSTANTS.DELETE_ASSESSMENT_REQUEST,
    assessmentId,
  }
}

export function updateSelectedAssessmentId(assessmentId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_ASSESSMENT_ID_REQUEST,
    assessmentId,
  }
}

export function fetchTemplates() {
  return {
    type: CONSTANTS.FETCH_TEMPLATES_REQUEST,
  }
}

export function addTemplate(template) {
  return {
    type: CONSTANTS.ADD_TEMPLATE_REQUEST,
    template,
  }
}

export function deleteTemplate(templateId) {
  return {
    type: CONSTANTS.DELETE_TEMPLATE_REQUEST,
    templateId,
  }
}

export function updateTemplate(templateId, template) {
  return {
    type: CONSTANTS.UPDATE_TEMPLATE_REQUEST,
    templateId,
    template,
  }
}

export function updateSelectedLearningStyleId(learningStyleId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_LEARNING_STYLE_ID_REQUEST,
    learningStyleId,
  }
}
export function updateSelectedPageId(pageId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_PAGE_ID_REQUEST,
    pageId,
  }
}

export function addKeyword(keyword) {
  return {
    type: CONSTANTS.ADD_KEYWORD_REQUEST,
    keyword,
  }
}

export function updateKeyword(keywordId, keyword) {
  return {
    type: CONSTANTS.UPDATE_KEYWORD_REQUEST,
    keywordId,
    keyword,
  }
}

export function deleteKeyword(keywordId) {
  return {
    type: CONSTANTS.DELETE_KEYWORD_REQUEST,
    keywordId,
  }
}

export function updateSelectedKeywordId(keywordId) {
  return {
    type: CONSTANTS.UPDATE_SELECTED_KEYWORD_ID_REQUEST,
    keywordId,
  }
}

export function updateReorderChallengeQuestion(questionId, text) {
  return {
    type: CONSTANTS.UPDATE_REORDER_CHALLENGE_QUESTION_REQUEST,
    questionId,
    text,
  }
}

export function addReorderChallengeStatement(statement, assessmentId) {
  return {
    type: CONSTANTS.ADD_REORDER_CHALLENGE_STATEMENT_REQUEST,
    statement,
    assessmentId,
  }
}

export function updateReorderChallengeStatement(
  statementId,
  statement,
  assessmentId,
) {
  return {
    type: CONSTANTS.UPDATE_REORDER_CHALLENGE_STATEMENT_REQUEST,
    statementId,
    statement,
    assessmentId,
  }
}

export function deleteReorderChallengeStatement(statementId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_REORDER_CHALLENGE_STATEMENT_REQUEST,
    statementId,
    assessmentId,
  }
}

export function updateCodeCompilingQuestion(questionId, question) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_QUESTION_REQUEST,
    questionId,
    text: question,
  }
}

export function updateCodeCompilingLanguage(languageId, language) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_LANGUAGE_REQUEST,
    languageId,
    text: language,
  }
}

export function updateCodeCompilingSolution(solutionId, solution) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_SOLUTION_REQUEST,
    solutionId,
    text: solution,
  }
}

export function updateCodeCompilingPlaceholder(placeholderId, placeholder) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_PLACEHOLDER_REQUEST,
    placeholderId,
    text: placeholder,
  }
}

export function updateCodeCompilingAddUseCase(assessmentId) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_ADD_TEST_CASE_REQUEST,
    assessmentId,
  }
}

export function updateCodeCompilingDeleteUseCase(testCaseId, assessmentId) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_DELETE_TEST_CASE_REQUEST,
    testCaseId,
    assessmentId,
  }
}

export function updateCodeCompilingUpdateUseCase(
  testCaseId,
  assessmentId,
  testCase,
) {
  return {
    type: CONSTANTS.UPDATE_CODE_COMPILING_UPDATE_TEST_CASE_REQUEST,
    testCaseId,
    assessmentId,
    testCase,
  }
}

export function addQuizQuestion(question, assessmentId) {
  return {
    type: CONSTANTS.ADD_QUIZ_QUESTION_REQUEST,
    question,
    assessmentId,
  }
}

export function updateQuizQuestion(questionId, question, assessmentId) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_QUESTION_REQUEST,
    questionId,
    question,
    assessmentId,
  }
}

export function deleteQuizQuestion(questionId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_QUIZ_QUESTION_REQUEST,
    questionId,
    assessmentId,
  }
}

export function updateQuizLanguage(languageId, language) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_LANGUAGE_REQUEST,
    languageId,
    text: language,
  }
}

export function updateQuizCode(codeId, code) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_CODE_REQUEST,
    codeId,
    text: code,
  }
}

export function addQuizChoice(choice, questionId, assessmentId) {
  return {
    type: CONSTANTS.ADD_QUIZ_CHOICE_REQUEST,
    choice,
    questionId,
    assessmentId,
  }
}

export function updateQuizChoice(choiceId, choice, questionId) {
  return {
    type: CONSTANTS.UPDATE_QUIZ_CHOICE_REQUEST,
    choiceId,
    choice,
    questionId,
  }
}

export function deleteQuizChoice(choiceId, questionId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_QUIZ_CHOICE_REQUEST,
    choiceId,
    questionId,
    assessmentId,
  }
}

export function addFibElement(element, assessmentId) {
  return {
    type: CONSTANTS.ADD_FIB_ELEMENT_REQUEST,
    element,
    assessmentId,
  }
}

export function updateFibElement(elementId, element, assessmentId) {
  return {
    type: CONSTANTS.UPDATE_FIB_ELEMENT_REQUEST,
    elementId,
    element,
    assessmentId,
  }
}

export function deleteFibElement(elementId, assessmentId) {
  return {
    type: CONSTANTS.DELETE_FIB_ELEMENT_REQUEST,
    elementId,
    assessmentId,
  }
}

export const loadSimulatedNodes = nodes => ({
  type: CONSTANTS.LOAD_SIMULATED_NODES_REQUEST,
  nodes,
})

export const setDidUpdate = didUpdate => ({
  type: CONSTANTS.SET_DID_UPDATE_REQUEST,
  didUpdate,
})