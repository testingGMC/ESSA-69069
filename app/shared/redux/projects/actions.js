/*
 *
 * Projects actions
 *
 */
import * as CONSTANTS from './constants'

export function fetchProjects() {
  return {
    type: CONSTANTS.FETCH_PROJECTS_REQUEST,
  }
}

export function addProject(project) {
  return {
    type: CONSTANTS.ADD_PROJECT_REQUEST,
    project,
  }
}
export function deleteProject(id) {
  return {
    type: CONSTANTS.DELETE_PROJECT_REQUEST,
    id,
  }
}
export function updateProject(id, project) {
  return {
    type: CONSTANTS.UPDATE_PROJECT_REQUEST,
    id,
    project,
  }
}
export const setIsVisible = isVisible => ({
  type: CONSTANTS.SET_IS_VISIBLE_REQUEST,
  isVisible,
})
export const setIsUpdateModalVisible = isVisible => ({
  type: CONSTANTS.SET_IS_UPDATE_MODAL_VISIBLE_REQUEST,
  isVisible,
})
