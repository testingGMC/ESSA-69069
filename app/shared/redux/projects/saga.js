import { put, takeEvery, all, call } from 'redux-saga/effects'
import * as api from 'shared/services/project.service'
import * as CONSTANTS from './constants'

export function* fetchProjects() {
  try {
    const data = yield call(api.fetchProjects)
    yield put({
      type: CONSTANTS.FETCH_PROJECTS_SUCCESS,
      projects: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.FETCH_PROJECTS_FAILURE, e })
  }
}

export function* addProject(action) {
  try {
    const data = yield call(api.addProject, action.project)
    yield put({
      type: CONSTANTS.ADD_PROJECT_SUCCESS,
      project: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.ADD_PROJECT_FAILURE, e })
  }
}
export function* deleteProject(action) {
  try {
    yield call(api.deleteProject, action.id)
    yield put({
      type: CONSTANTS.DELETE_PROJECT_SUCCESS,
      projectId: action.id,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.DELETE_PROJECT_FAILURE, e })
  }
}

export function* setIsVisible({ isVisible }) {
  try {
    yield put({
      type: CONSTANTS.SET_IS_VISIBLE_SUCCESS,
      isVisible,
    })
  } catch (e) {
    yield put({
      type: CONSTANTS.SET_IS_VISIBLE_FAILURE,
    })
  }
}
export function* updateProject(action) {
  try {
    const data = yield call(api.updateProject, action.id, action.project)
    yield put({
      type: CONSTANTS.UPDATE_PROJECT_SUCCESS,
      project: data.model,
    })
  } catch (e) {
    yield put({ type: CONSTANTS.UPDATE_PROJECT_FAILURE, e })
  }
}
export function* setIsUpdateModalVisible({ isVisible }) {
  try {
    yield put({
      type: CONSTANTS.SET_IS_UPDATE_MODAL_VISIBLE_SUCCESS,
      isVisible,
    })
  } catch (e) {
    yield put({
      type: CONSTANTS.SET_IS_UPDATE_MODAL_VISIBLE_FAILURE,
    })
  }
}


export function* setIsVisibleWatcher() {
  yield takeEvery(CONSTANTS.SET_IS_VISIBLE_REQUEST, setIsVisible)
}

export function* fetchProjectsWatcher() {
  yield takeEvery(CONSTANTS.FETCH_PROJECTS_REQUEST, fetchProjects)
}
export function* addProjectWatcher() {
  yield takeEvery(CONSTANTS.ADD_PROJECT_REQUEST, addProject)
}
export function* deleteProjectWatcher() {
  yield takeEvery(CONSTANTS.DELETE_PROJECT_REQUEST, deleteProject)
}
export function* updateProjectWatcher() {
  yield takeEvery(CONSTANTS.UPDATE_PROJECT_REQUEST, updateProject)
}
export function* setIsUpdateModalVisibleWatcher() {
  yield takeEvery(
    CONSTANTS.SET_IS_UPDATE_MODAL_VISIBLE_REQUEST,
    setIsUpdateModalVisible,
  )
}

export default function* selectedProjectsSaga() {
  yield all([
    fetchProjectsWatcher(),
    addProjectWatcher(),
    deleteProjectWatcher(),
    updateProjectWatcher(),
    setIsVisibleWatcher(),
    setIsUpdateModalVisibleWatcher(),
  ])
}
