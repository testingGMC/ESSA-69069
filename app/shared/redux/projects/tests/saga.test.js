
import { put, takeEvery } from 'redux-saga/effects'
import * as sagas from '../saga'
import * as actions from '../actions'
import * as CONSTANTS from '../constants'

describe('fetchProjects Saga', () => {
  let fetchProjectsGenerator
  const projectId = '1'
  const action = actions.fetchProjects(projectId)

  beforeEach(() => {
    fetchProjectsGenerator = sagas.fetchProjects(action)
    const callDescriptor = fetchProjectsGenerator.next().value
    expect(callDescriptor).toMatchSnapshot()
  })

  it('should dispatch the FETCH_PROJECTS_SUCCESS action if it requests the data successfully', () => {
    const response = {
      model: [
        { id: '1', name: 'project 1' },
        { id: '2', name: 'project 2' },
        { id: '3', name: 'project 3' },
      ],
    }
    const successAction = {
      type: CONSTANTS.FETCH_PROJECTS_SUCCESS,
      projects: response.model,
    }
    const putDescriptor = fetchProjectsGenerator.next(response).value
    expect(putDescriptor).toEqual(put(successAction))
  })

  it('should call the FETCH_PROJECTS_FAILURE action if the response errors', () => {
    const response = new Error('Some error')
    const putDescriptor = fetchProjectsGenerator.throw(response).value
    const failureAction = {
      type: CONSTANTS.FETCH_PROJECTS_FAILURE,
    }
    expect(putDescriptor).toEqual(put(failureAction))
  })
})


describe('fetchProjectsWatcher Saga', () => {
  const fetchProjectsWatcher = sagas.fetchProjectsWatcher()

  it('should start task to watch for FETCH_PROJECTS_REQUEST action', () => {
    const takeEveryDescriptor = fetchProjectsWatcher.next().value

    expect(takeEveryDescriptor).toEqual(
      takeEvery(CONSTANTS.FETCH_PROJECTS_REQUEST, sagas.fetchProjects),
    )
  })
})
