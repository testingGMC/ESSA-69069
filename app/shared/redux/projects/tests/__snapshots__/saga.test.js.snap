// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchProjects Saga should call the FETCH_PROJECTS_FAILURE action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`fetchProjects Saga should dispatch the FETCH_PROJECTS_SUCCESS action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
